src/sdk/sql_cluster_router.cc

```diff
@@ -865,38 +944,58 @@ std::shared_ptr<hybridse::sdk::ResultSet> SQLClusterRouter::ExecuteSQLBatchReque
     if (status == NULL) return false;
     std::shared_ptr<::openmldb::nameserver::TableInfo> table_info;
-    DefaultValueMap default_map;
-    uint32_t str_length = 0;
-    if (!GetInsertInfo(db, sql, status, &table_info, &default_map, &str_length)) {
+    std::vector<DefaultValueMap> default_maps;
+    std::vector<uint32_t> str_lengths;
+    if (!GetMultiRowInsertInfo(db, sql, status, &table_info, &default_maps, &str_lengths)) {
         status->code = 1;
         LOG(WARNING) << "get insert information failed";
         return false;
     }
-    std::shared_ptr<SQLInsertRow> row = std::make_shared<SQLInsertRow>(
-        table_info, ::openmldb::sdk::ConvertToSchema(table_info), default_map, str_length);
-    if (!row) {
-        status->msg = "fail to parse row from sql " + sql;
-        LOG(WARNING) << status->msg;
-        return false;
-    }
-    if (!row->Init(0)) {
-        status->msg = "fail to encode row for table " + table_info->name();
+
+    if (default_maps.empty() || str_lengths.empty() || default_maps.size() != str_lengths.size()) {
+        status->msg = "default maps is empty or str_lengths are empty or default_maps";
+        status->code = 1;
         LOG(WARNING) << status->msg;
         return false;
     }
-    if (!row->IsComplete()) {
-        status->msg = "insert value isn't complete";
+    if (default_maps.size() != str_lengths.size()) {
+        status->msg = "default maps isn't match with str_lengths";
+        status->code = 1;
         LOG(WARNING) << status->msg;
         return false;
     }
-    std::vector<std::shared_ptr<::openmldb::catalog::TabletAccessor>> tablets;
-    bool ret = cluster_sdk_->GetTablet(db, table_info->name(), &tablets);
-    if (!ret || tablets.empty()) {
-        status->msg = "fail to get table " + table_info->name() + " tablet";
-        LOG(WARNING) << status->msg;
-        return false;
+    std::shared_ptr<::hybridse::sdk::Schema> schema = ::openmldb::sdk::ConvertToSchema(table_info);
+
+    for (int i = 0; i < default_maps.size(); i++) {
+        std::shared_ptr<SQLInsertRow> row = std::make_shared<SQLInsertRow>(table_info, schema, default_maps[i],
+                                                                           str_lengths[i]);
+        if (!row) {
+            status->msg = "fail to parse row from sql " + sql;
+            LOG(WARNING) << status->msg;
+            return false;
+        }
+        if (!row->Init(0)) {
+            status->msg = "fail to encode row for table " + table_info->name();
+            LOG(WARNING) << status->msg;
+            return false;
+        }
+        if (!row->IsComplete()) {
+            status->msg = "insert value isn't complete";
+            LOG(WARNING) << status->msg;
+            return false;
+        }
+        std::vector<std::shared_ptr<::openmldb::catalog::TabletAccessor>> tablets;
+        bool ret = cluster_sdk_->GetTablet(db, table_info->name(), &tablets);
```

Comment:
Perhaps ```GetTablet``` function should be outside the loop section, i don't know whether there is load-balance algorithm in this function so we should call it before ```PutRow```.

Reply:
Actually, there isn't a dynamic load-balance algorithm here. So we can simply put `GetTablet` out of the loop. Just a bad design. I will fix it.